src/test/java/com/zaxxer/nuprocess/CatTest.java

```diff
@@ -335,6 +337,7 @@ public void softCloseStdinAfterWrite() throws Exception
          this.writeAdler32 = new Adler32();
 
          bytes = getLotsOfBytes();
+         exitCode = -1;
```

Comment:
Without this, the exit code on the handler defaults to `0`. That means if `onExit` is _never called_, the `getExitCode() == 0` checks in the various tests will still pass. By setting this to `-1` (which, I'll add, `Utf8DecodingListener` already does), if `onExit` is never called the tests will fail.